modules/nf-commons/src/main/nextflow/util/MemoryUnit.groovy

```diff
@@ -95,6 +95,17 @@ class MemoryUnit implements Comparable<MemoryUnit>, Serializable, Cloneable {
 
     }
 
+    /**
+    * Function to parse/convert given memory unit 
+    *
+    * @param unit String expressing memory unit in bytes, e.g. KB, MB, GB
+    */
+    long toUnit(String unit){
+        int p = UNITS.indexOf(unit)
+        return Math.round( size/Math.pow(1024, p) )
+    }
+
```

Comment:
Good, tho it should be checked in p==-1 (not found) and throwing an exception e.g.  `throw new IllegalAgumentException("Not a valid memory unit: $unit")`

Reply:
Very good point; I was wondering if there were some checks above.

Reply:
OK, consider that 

```groovy
        if (p==-1){
            throw new IllegalArgumentException("Not a valid memory unit in bytes: $unit")
        } else{
            return Math.round( size/Math.pow(1024, p) )
        }
```

can be replace by 

```
        if (p==-1)
            throw new IllegalArgumentException("Not a valid memory unit in bytes: $unit")
        return Math.round( size/Math.pow(1024, p) )

Reply:
Good point! 

I do like brackets---either 

```
if (p==-1)
    throw new IllegalArgumentException("Not a valid memory unit in bytes: $unit")
return Math.round( size/Math.pow(1024, p) )
```
or

```
if (p==-1){
    throw new IllegalArgumentException("Not a valid memory unit in bytes: $unit")
}
return Math.round( size/Math.pow(1024, p) )

```

works for me